eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -980,6 +980,41 @@ public function testUpdateLocationStructValues(array $data)
      *
      * @see \eZ\Publish\API\Repository\LocationService::updateLocation()
      * @depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testLoadLocation
+     */
+    public function testUpdateLocationWithSameRemoteId()
+    {
+        $repository = $this->getRepository();
+
+        $locationId = $this->generateId('location', 5);
+        /* BEGIN: Use Case */
+        // $locationId and remote ID is the IDs of the same, existing location
+        $locationService = $repository->getLocationService();
+
+        $originalLocation = $locationService->loadLocation($locationId);
+
+        $updateStruct = $locationService->newLocationUpdateStruct();
+
+        // Remote ID of an existing location with the same locationId
+        $updateStruct->remoteId = '3f6d92f8044aed134f32153517850f5a';
+
+        // Sets some random priority to confirm the update
+        $updateStruct->priority = 2;
+
+        $location = $locationService->updateLocation($originalLocation, $updateStruct);
+
+        // Checks that the location was updated
+        $this->assertEquals(2, $location->priority);
+
+        // Checks that remoteId remains the same
+        $this->assertEquals($originalLocation->remoteId, $location->remoteId);
+        /* END: Use Case */
```

Comment:
Only to make sure that remote ID remains the same, that's the whole point of the test. Otherwise, if remote ID does not remain the same (because storage engine maybe changes it internally for example), the test does not make sense, right?